### ISSUE TITLE
the selection of T5-XXL model

### ISSUE THREAD

**Issue by Amazon90** [city96/ComfyUI-GGUF#58]
If I download `flux1-dev-Q8_0.gguf`, does that mean I need to download `t5-v1_1-xxl-encoder-Q8_0.gguf`, and should I avoid the Q6 or f16 versions?
Alternatively, if I download `flux1-dev-Q8_0.gguf`, can the T5-XXL model utilize `t5xxl_fp8_e4m3fn.safetensors`?
Additionally, after downloading `flux1-dev-Q8_0.gguf`, do I also need to download the VAE model `ae.safetensors`?

**Comment by city96** [comment]
You can mix and match the T5 and model precisions, even the original FP16 one is fine if you can load it. The VAE/CLIP model should just be the defaults so yes, you still need those.

**Comment by Amazon90** [comment]
> You can mix and match the T5 and model precisions, even the original FP16 one is fine if you can load it. The VAE/CLIP model should just be the defaults so yes, you still need those.

I understand now. So, `flux1-dev-Q8_0.gguf` and `flux1-dev-bnb-nf4.safetensors` are not the same. The `flux1-dev-bnb-nf4.safetensors` model integrates CLIP, T5-XXL, and VAE into the model, while `flux1-dev-Q8_0.gguf`, like `flux1-dev-fp8.safetensors`, is a compressed version of the original `flux1-dev.safetensors` model. To use it, I will need to download the CLIP, T5-XXL, and VAE models separately.

**Comment by city96** [comment]
Yup, you got it.

**Comment by Amazon90** [comment]
> Yup, you got it.

Awesome, thank you for your guidance and for the work you’ve put into this. It allows me to enjoy the amazing experience that flux provides.

**Comment by Amazon90** [comment]
> Yup, you got it.

I'm sorry, but after using it, I encountered some new issues, and I hope you can help me with three questions:

1. I placed the file `t5-v1_1-xxl-encoder-Q8_0.gguf` in the `ComfyUI\models\clip` folder, but I cannot select it in the clip loader.
![1](https://github.com/user-attachments/assets/1e6856ba-1ba8-4c50-bc43-2286ce381b00)

2. The image generation speed of `flux1-dev-Q8_0.gguf` is only faster than `flux1-dev-fp8.safetensors` during the first image generation, taking 66 seconds for the GGUF model and 106 seconds for the FP8 model. However, afterward, the GGUF model takes around 40 seconds, while the FP8 model only takes about 30 seconds. My graphics card is a 4070 Ti with 12GB of memory.
![2](https://github.com/user-attachments/assets/37ef0b3e-2308-4be5-8046-badb9d5fa346)

3. I find it strange that the generated images are not clear, regardless of whether I use the FP8 or GGUF model. They appear blurred, as if a Gaussian blur effect has been applied.
![3](https://github.com/user-attachments/assets/8b93b85b-025f-46d7-8b54-0746e7aa4e86)

**Comment by city96** [comment]
>1 ...

There is a separate "DualClipLoader (GGUF)" node that's needed to load it, just like with the UNET. I haven't found a clean way to just use the built in ones without breaking things.

>2 ...

This is expected. The first run is usually irrelevant since it has to load the model from disk. After that, the GGUF files will be slower since your GPU has to do more math to use them as they're a more advanced format. The benefit is mostly in Q8_0 being closer to FP16 than FP8 is, as well as being able to quantize even further for people with less vram.

I'll get around to optimizing the speed eventually.

>3 ...

Change the resolution to 768x1280. Some resolutions behave weirdly due to interpolation.

**Comment by Amazon90** [comment]
> >1 ...
> 
> 
> 
> There is a separate "DualClipLoader (GGUF)" node that's needed to load it, just like with the UNET. I haven't found a clean way to just use the built in ones without breaking things.
> 
> 
> 
> >2 ...
> 
> 
> 
> This is expected. The first run is usually irrelevant since it has to load the model from disk. After that, the GGUF files will be slower since your GPU has to do more math to use them as they're a more advanced format. The benefit is mostly in Q8_0 being closer to FP16 than FP8 is, as well as being able to quantize even further for people with less vram.
> 
> 
> 
> I'll get around to optimizing the speed eventually.
> 
> 
> 
> >3 ...
> 
> 
> 
> Change the resolution to 768x1280. Some resolutions behave weirdly due to interpolation.

Thank you for your answer, master! I can't believe that simply changing the image resolution can solve the problem. I would never have thought of that, no matter how hard I tried!

**Comment by green-anger** [comment]
> Change the resolution to 768x1280. Some resolutions behave weirdly due to interpolation.

@city96  An off topic question, but since you mentioned the resolution. Is there any source of recommended and supported image resolutions? We have a list for SDXL in the paper and the one used in the example above fits it perfectly (896x1152). But I didn't find one in Flux paper (maybe it was a wrong one or I didn't look carefully), nothing in their github or HF repo. Why did you suggest 768x1280?

**Comment by Amazon90** [comment]
> > Change the resolution to 768x1280. Some resolutions behave weirdly due to interpolation.
> 
> 
> 
> @city96  An off topic question, but since you mentioned the resolution. Is there any source of recommended and supported image resolutions? We have a list for SDXL in the paper and the one used in the example above fits it perfectly (896x1152). But I didn't find one in Flux paper (maybe it was a wrong one or I didn't look carefully), nothing in their github or HF repo. Why did you suggest 768x1280?

https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/1157#issuecomment-2291866572

**Comment by green-anger** [comment]
> > @city96  An off topic question, but since you mentioned the resolution. Is there any source of recommended and supported image resolutions? We have a list for SDXL in the paper and the one used in the example above fits it perfectly (896x1152). But I didn't find one in Flux paper (maybe it was a wrong one or I didn't look carefully), nothing in their github or HF repo. Why did you suggest 768x1280?
> 
> [lllyasviel/stable-diffusion-webui-forge#1157 (comment)](https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/1157#issuecomment-2291866572)

Doesn't answer the question at all.

**Comment by city96** [comment]
@green-anger Sorry I missed this. That was a complete guess.

I know transformer models do worse with resolutions they weren't trained on (compared to conv2d/unet models), and just figured that a common 16:9 bucket would've had plenty of examples for good training data compared to a bucket like 1152x896, which it might have seen less of during training.

Flux uses RoPE for the positional embeddings, which is the reason it can generate at most resolutions without falling apart (you can look up LLM context extending and RoPE scaling to read more about this), but that still causes weirdness with some resolutions due to the interpolation not lining up - maybe they didn't have 1152x896 as a bucket during training?

haven't seen any paper/pdf/etc by black forest labs so we don't really have anything to go off on. Using beta/normal instead of "simple" as the scheduler is definitely recommended though, and should make the effect less visible.

**Comment by Amazon90** [comment]
> all.

I later found that even when using an image resolution of 896x1152, if I select Euler as the sampler and use beta instead of simple for the scheduler, the image results become clearer and are no longer blurry. This is because, in my experience, sometimes even when the resolution is changed to 768x1280, the images can still appear blurry.

**Comment by city96** [comment]
@green-anger Please see the explanation here: https://github.com/city96/ComfyUI-GGUF/issues/36#issuecomment-2295300470
